src/components/AssignmentSummary.jsx

```diff
@@ -117,7 +117,7 @@ export class AssignmentSummary extends Component {
               title: 'Send first texts',
               count: unmessagedCount,
               primary: true,
-              disabled: (useDynamicAssignment && !hasUnassignedContacts && unmessagedCount == 0) || (useDynamicAssignment && maxContacts === 0),
+              disabled: (useDynamicAssignment && !hasUnassignedContacts && unmessagedCount == 0 && maxContacts !== 0) ? true : false,
```

Comment:
It might make sense to move this boolean logic to L14, before `this.renderBadgedButton` is called, so it can be described more clearly. Also, is the `? true : false` necessary here?

Reply:
agree.  I also think the `&& maxContacts != 0` is maybe incorrect.  If maxContacts is not zero, it's still possible that it's disabled, right?  The logic might be a bit weirder....

Reply:
thanks for the feedback. i think this part doesnt need to be changed after doing further qa